nucleus/payara-modules/healthcheck-core/src/main/java/fish/payara/nucleus/healthcheck/preliminary/GarbageCollectorHealthCheck.java

```diff
@@ -92,52 +98,114 @@ protected String getDescription() {
+        for (GcUsage u : collect.values()) {
+            timePassed = Math.max(timePassed, u.getTimePassed());
+            timeSpendDoingGc += u.getTimeSpendDoingGc();
+            numberOfGcs += u.getNumberOfGcs();
+        }
+        collectGcUage(collector, "Total", 100d * timeSpendDoingGc / timePassed, numberOfGcs, timeSpendDoingGc);
+    }
 
-                if (diffTime > 0 && youngLastCollectionCount > 0) {
-                    percentage = ((diffCount) / (youngLastCollectionCount)) * 100;
+    private static void collectGcUage(MonitoringDataCollector collector, String label, double usage, long numberOfGcs, long timeSpendDoingGc) {
+        collector
+            .collect(label + "GcPercentage", (long) usage)
+            .collect(label + "GcCount", numberOfGcs)
+            .collect(label + "GcDuration", timeSpendDoingGc);
+    }
 
-                    result.add(new HealthCheckResultEntry(decideOnStatusWithRatio(percentage),
-                            diffCount + " times Young GC (" + gcBean.getName()  + ") after " + prettyPrintDuration(diffTime)));
-                }
+    private static final class GcUsage {
 
-                youngLastCollectionCount = gcBean.getCollectionCount();
-                youngLastCollectionTime = gcBean.getCollectionTime();
-            }
-            else if (OLD_PS_MARKSWEEP.equals(gcBean.getName()) ||
-                    OLD_G1GC.equals(gcBean.getName()) ||
-                    OLD_MARK_SWEEP_COMPACT.equals(gcBean.getName()) ||
-                    OLD_CONCURRENTMARKSWEEP.equals(gcBean.getName())) {
-                long diffCount = gcBean.getCollectionCount() - oldLastCollectionCount;
-                long diffTime = gcBean.getCollectionTime() - oldLastCollectionTime;
-
-                if (diffTime > 0 && oldLastCollectionCount > 0) {
-                    percentage = ((diffCount) / (oldLastCollectionCount)) * 100;
-
-                    result.add(new HealthCheckResultEntry(decideOnStatusWithRatio(percentage),
-                            diffCount + " times Old GC (" + gcBean.getName()  + ") after " + prettyPrintDuration(diffTime)));
-                }
-
-                oldLastCollectionCount = gcBean.getCollectionCount();
-                oldLastCollectionTime = gcBean.getCollectionTime();
-            }
-            else {
-                result.add(new HealthCheckResultEntry(HealthCheckResultStatus.CHECK_ERROR, "Could not identify " +
-                        "GarbageCollectorMXBean with name: " + gcBean.getName()));
-            }
+        private volatile long timeLastChecked;
+        private volatile long lastCollectionCount;
+        private volatile long lastCollectionTime;
+        private volatile long timeSpendDoingGc;
+        private volatile long numberOfGcs;
+        private volatile long timePassed;
+
+        GcUsage() {
+            // make visible
         }
 
-        return result;
+        public double percentage(GarbageCollectorMXBean gcBean) {
```

Comment:
NB: This logic is intentionally not the same as the one replaced.

See #3291 for more details.